csharp/extractor/Semmle.Extraction.CSharp/Extractor/Extractor.cs

```diff
@@ -92,16 +94,63 @@ public static ILogger MakeLogger(Verbosity verbosity, bool includeConsole)
         /// <param name="args">Command line arguments as passed to csc.exe</param>
         /// <returns><see cref="ExitCode"/></returns>
         public static ExitCode Run(string[] args)
+        {
+            var options = Options.CreateWithEnvironment(args);
+            if (options.BinaryLogPath is string binlogPath)
+            {
+                using var fileStream = new FileStream(binlogPath, FileMode.Open, FileAccess.Read, FileShare.Read);
+
+                // Filter out compiler calls that aren't interesting for examination
+                var predicate = bool (CompilerCall compilerCall) =>
+                {
+                    if (!compilerCall.IsCSharp)
+                    {
+                        return false;
+                    }
+
+                    return compilerCall.Kind switch
+                    {
+                        CompilerCallKind.XamlPreCompile => false,
+                        CompilerCallKind.Satellite => false,
+                        CompilerCallKind.WpfTemporaryCompile => false,
+                        _ => true
+                    };
+                };
+
+                var compilerCalls = BinaryLogUtil.ReadAllCompilerCalls(fileStream, predicate);
+                var exitCode = ExitCode.Ok;
+                foreach (var compilerCall in compilerCalls)
+                {
+                    var diagnosticName = compilerCall.GetDiagnosticName();
+                    Console.WriteLine($"Processing {diagnosticName}");
+                    var compilerCallOptions = Options.CreateWithEnvironment([]);
+                    compilerCallOptions.CompilerName = compilerCall.CompilerFilePath;
+                    compilerCallOptions.CompilerArguments.AddRange(compilerCall.GetArguments());
+                    using var logger = MakeLogger(compilerCallOptions.Verbosity, compilerCallOptions.Console);
+                    logger.LogInfo($"Running extractor on arguments from binary log. Processing {diagnosticName}.");
+                    var ec = Run(compilerCallOptions, logger);
+                    if (ec != ExitCode.Ok)
+                    {
+                        exitCode = ec;
```

Comment:
This is probably not okay. We should return `Failed` only if all compilation calls failed.